compiler/AST/symbol.cpp

```diff
@@ -2275,6 +2356,92 @@ bool FnSymbol::isSecondaryMethod() const {
   return isMethod() && !isPrimaryMethod();
 }
 
+
+// This function or method is an iterator (as opposed to a procedure).
+bool FnSymbol::isIterator() const {
+  return hasFlag(FLAG_ITERATOR_FN);
+}
+
+
+std::string FnSymbol::docsDirective() {
+  if (fDocsTextOnly) {
+    return "";
+  }
+
+  if (this->isMethod() && this->isIterator()) {
+    return ".. itermethod:: ";
+  } else if (this->isIterator()) {
+    return ".. iterfunction:: ";
+  } else if (this->isMethod()) {
+    return ".. method:: ";
+  } else {
+    return ".. function:: ";
+  }
+}
+
+
+void FnSymbol::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->hasFlag(FLAG_NO_DOC)) {
+    return;
+  }
+
+  // Print the rst directive, if one is needed.
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+
+  // Print inline/export.
+  if (this->hasFlag(FLAG_INLINE)) {
+    *file << "inline ";
+  } else if (this->hasFlag(FLAG_EXPORT)) {
+    *file << "export ";
+  }
+
+  // Print iter/proc.
+  if (this->isIterator()) {
+    *file << "iter ";
+  } else {
+    *file << "proc ";
+  }
+
+  // Print name and arguments.
+  AstToText *info = new AstToText();
+  info->appendNameAndFormals(this);
+  *file << info->text();
+  delete info;
+
+  // Print ref return intent, if one exists.
```

Comment:
Let's drop the "ref" part of the comment, since we are printing any return intent.